frame/staking/src/tests.rs

```diff
@@ -3957,7 +3958,7 @@ mod election_data_provider {
 	#[test]
 	fn respects_len_limits() {
 		ExtBuilder::default().build_and_execute(|| {
-			assert_eq!(Staking::voters(Some(1)).unwrap_err(), "Voter snapshot too big");
+			assert_eq!(Staking::voters(Some(1)).unwrap().0.len(), 1);
```

Comment:
i dont understand what this assertion tells us

Reply:
`Staking::voters`' parameter is a `maybe_length_limit`. The previous behavior when a limit was specified, but the number of actual voters exceeded the limit, was to raise an error.

This PR changes that behavior: instead of raising an error, it now just produces only the requested number of voters in case a limit is present. 

This assertion is that, given more than one actual voter, but a requested limit of 1, the actual number of produced voters is 1.